bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVMStructs.java

```diff
@@ -104,9 +104,11 @@ private static BValue getBValue(BType type, Object value) {
                 break;
             case TypeTags.BYTE_TAG:
                 if (value instanceof Byte) {
-                    return new BByte(((Byte) value));
+                    return new BByte(((Byte) value).longValue());
                 } else if (value instanceof Integer) {
-                    return new BByte(((Integer) value).byteValue());
+                    return new BByte(((Integer) value).longValue());
```

Comment:
@pranavan15 will this guarantee the byte overflow?

Reply:
I have updated this logic with commit 51d1715 to handle possible byte overflow and underflow.